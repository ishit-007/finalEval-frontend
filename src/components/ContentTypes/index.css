.content-types {
  height: 100vh;
  background-color: blue;
  width: 70vw;
}
.content-types .heading {
  margin-bottom: 0;
  background-color: white;
  color: #414041;
  height: 8%;
  padding-top: 20px;
  padding-left: 20px;
}
.new-content {
  display: flex;
  flex-direction: column;
}
.new-content input {
  width: 500px;
}
.modal-content,
.modal-content2 {
  border-radius: 10px;
  background-color: #fefefe;
  margin: 15% auto; /* 15% from the top and centered */
  padding: 40px;
  border: 1px solid #888;
  width: 35%; /* Could be more or less, depending on screen size */
}
.new-content p {
  margin-top: 20px;
}
.modal-content input,
.modal-content2 input {
  width: 90%;
  margin-top: 20px;
  height: 40px;
  border-radius: 5px;
  border: 1px solid rgb(18, 154, 252);
}
.btns {
  gap: 1rem;
  margin-top: 25px;
  width: 100%;
  display: flex;
  justify-content: flex-end;
}
.btns .cancel {
  background-color: inherit;
  border: none;
  padding: 20px;
  font-weight: 700;
  font-size: medium;
}
.btns .create {
  font-weight: 700;
  font-size: medium;
  background-color: #5a03ce;
  border: none;
  color: white;
  padding-left: 20px;
  padding-right: 20px;
  width: 200px;
  border-radius: 6px;
}
.modal,
.modal2 {
  display: none; /* Hidden by default */
  position: fixed; /* Stay in place */
  z-index: 1; /* Sit on top */
  padding-top: 100px; /* Location of the box */
  left: 0;
  top: 0;
  width: 100%; /* Full width */
  height: 100%; /* Full height */
  overflow: auto; /* Enable scroll if needed */
  background-color: rgb(0, 0, 0);
  background-color: rgba(0, 0, 0, 0.4);
}
.existing {
  border-radius: 6px;
  color: #939393;
  padding: 10px;
  margin-top: 20px;
  display: flex;
  justify-content: space-between;
  width: 80%;
  margin-left: 10%;
  height: 45px;
  align-content: center;
  background-color: white;
  font-weight: 600;
}
.addNewType {
  font-weight: 600;
  margin-left: 10%;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 80%;
  height: 50px;
  background-color: white;
  color: #5a03ce;
  border-color: #5a03ce;
  border-style: dashed;
  border-radius: 6px;
}
.container {
  background-color: #ebeefe;
  height: 92%;
}
.one-type-fields {
  position: absolute;
  right: 0;
  top: 8vh;
  background-color: #ebeefe;
  width: 49%;
  height: 92%;
}
.one-type-fields {
  background-color: #ebeefe;
}
.one-type-fields .heading {
  padding: 20px;
  background-color: #ebeefe;
}
.one-type-fields .heading h3 {
  padding-top: 10px;
  color: #95969e;
}
.container .new-types-content {
  height: 92%;
}
.container .new-types {
  width: 30%;
  height: 100%;
  background-color: #d4d6e5;
}
.container .new-types-header {
  padding-left: 35px;
  padding-top: 20px;
  padding-bottom: 20px;
  padding-right: 35px;
  background-color: #d4d6e5;
  width: 30%;
  display: flex;
  justify-content: space-between;
}
.addField {
  font-weight: 600;
  color: #5a03ce;
  height: 40px;
  background-color: #f4f6ff;
  border-radius: 6px;
  width: 80%;
  margin-top: 30px;
  margin-left: 10%;
  display: flex;
  justify-content: center;
  align-items: center;
  border: 1px #5a03ce;
  border-style: dashed;
}
.existing-field img {
  margin-right: 10px;
}
.existing-field {
  color: #aaaaaa;
  border-radius: 4px;
  margin-top: 20px;
  height: 40px;
  margin-left: 10%;
  background-color: white;
  width: 80%;
  display: flex;
  justify-content: space-between;
  align-items: center;
}
.existing-field .logo {
  display: flex;
  align-items: center;
  justify-content: center;
  font-weight: 700;
  color: white;
  border-radius: 4px;
  width: 10%;
  background-color: #7691ff;
  height: 40px;
}
