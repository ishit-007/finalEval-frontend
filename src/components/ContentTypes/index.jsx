/* eslint-disable react/jsx-key */
/* eslint-disable no-unused-vars */
import './index.css';
import React from 'react';
import magnifyingGlass from '../../assets/icon-search-dark.png';
import deleteButton from '../../assets/trash-delete-recycle-bin-bucket-waste.png';
import editButton from '../../assets/user-edit-text-message-note.png';
import axios from 'axios';
import PropTypes from 'prop-types';
// import { NewTypes } from '../../components';

const ContentTypes = props => {
  const [selectedContentType, setSelectedContentType] = React.useState('CompanyProfile');

  const [modal2IsOpen, setModal2IsOpen] = React.useState('none');
  const [selectedContentData, setSelectedContentData] = React.useState([]);
  const [newContentType, setNewContentType] = React.useState('');
  const [modalIsOpen, setModalIsOpen] = React.useState('none');
  const [contentTypes, setContentTypes] = React.useState([]);
  const [newField, setNewField] = React.useState('');

  const createNewContentTypeHandler = e => {
    axios
      .post('http://localhost:8080/contentTypes/create', {
        contentTypeName: newContentType,
      })
      .then(resp => {
        setModalIsOpen('none');
        setModal2IsOpen('none');
        setNewContentType('');
        window.location.reload();
      })
      .catch(err => {
        console.log(err);
      });
  };
  const createNewFieldHandler = e => {
    axios
      .patch('http://localhost:8080/attributes/create', {
        contentTypeName: selectedContentType,
        attribute: newField,
      })
      .then(resp => {
        setModal2IsOpen('none');
        setNewField('');
        window.location.reload();
      })
      .catch(err => {
        console.log(err);
      });
  };
  const deleteFieldHandler = attribute => {
    console.log(attribute, selectedContentData.id);
    axios
      .delete(`http://localhost:8080/attributes/delete/${selectedContentData.id}/${attribute}`, )
      .then(resp => {
        console.log(resp);
        window.location.reload();
      })
      .catch(err => {
        console.log(err);
      });
  };
  window.onclick = e => {
    if (e.target.className === 'modal') {
      setModalIsOpen('none');
      setNewContentType('');
    } else if (e.target.className === 'modal2') {
      setModal2IsOpen('none');
    }
  };

  React.useEffect(() => {
    axios
      .get('http://localhost:8080/contentTypes/all')
      .then(resp => {
        setContentTypes(resp.data);
        const filteredData = resp.data.filter(contentType => {
          console.log('selectedContentType: ', selectedContentType);
          if (contentType.name === selectedContentType) {
            return true;
          } else {
            return false;
          }
        });
        setSelectedContentData(filteredData[0]);
      })
      .catch(err => {
        console.log(err);
      });
  }, [selectedContentType]);

  return (
    <div className="content-types">
      <div className="heading">
        <h1
          onClick={() => {
            console.log(selectedContentData);
          }}>
          Content Types
        </h1>
      </div>
      <div className="modal" style={{ display: modalIsOpen }}>
        <div className="modal-content">
          <h3>Create a New Content Type</h3>

          <div className="new-content">
            <p>Name of the content type</p>
            <input
              type="text"
              value={newContentType}
              onChange={event => {
                setNewContentType(event.target.value);
              }}
            />
          </div>
          <div className="btns">
            <button
              className="cancel"
              onClick={() => {
                setModalIsOpen('none');
              }}>
              Cancel
            </button>
            <button className="create" onClick={createNewContentTypeHandler}>
              Create
            </button>
          </div>
        </div>
      </div>
      <div className="modal2" style={{ display: modal2IsOpen }}>
        <div className="modal-content2">
          <h3>Create a New Field</h3>

          <div className="new-content">
            <p>Name of the field</p>
            <input
              type="text"
              value={newField}
              onChange={event => {
                setNewField(event.target.value);
              }}
            />
          </div>
          <div className="btns">
            <button className="cancel">Cancel</button>
            <button className="create" onClick={createNewFieldHandler}>
              Create
            </button>
          </div>
        </div>
      </div>
      <div className="container">
        <div className="new-types-content">
          <div className="new-types-header">
            <p>{contentTypes && contentTypes.length} Types</p>
            <img src={magnifyingGlass} alt="" />
          </div>
          <div className="new-types">
            <div
              className="addNewType"
              onClick={() => {
                setModalIsOpen('block');
              }}>
              + New Type
            </div>
            <div className="existing-types">
              {contentTypes.map(contentType => {
                return (
                  <div
                    key={Math.random()}
                    className="existing"
                    onClick={() => {
                      setSelectedContentType(contentType.name);
                    }}>
                    <p>{contentType.name}</p>
                    <p>13</p>
                  </div>
                );
              })}
            </div>
          </div>
          <div className="single-type-fields"></div>
        </div>
        <div className="one-type-fields">
          <div className="heading">
            <h1>Company_Profile</h1>
            <h3>
              {selectedContentData && selectedContentData.attributes && selectedContentData.attributes.length} Fields
            </h3>
          </div>
          <div
            className="addField"
            onClick={() => {
              setModal2IsOpen('block');
            }}>
            Add another field
          </div>
          <div className="existing-fields">
            {selectedContentData &&
              selectedContentData.attributes &&
              selectedContentData.attributes.map(attribute => {
                if (attribute) {
                  return (
                    <div className="existing-field">
                      <div className="logo">Ab</div>
                      <div>{attribute}</div>
                      <div>Text</div>
                      <div className="icons">
                        <img
                          src={deleteButton}
                          alt=""
                          style={{ height: '20px', width: '20px' }}
                          onClick={() => {
                            deleteFieldHandler(attribute);
                          }}
                        />
                        <img src={editButton} alt="" style={{ height: '20px', width: '20px' }} />
                      </div>
                    </div>
                  );
                }
              })}
          </div>
        </div>
      </div>
    </div>
  );
};
// ContentTypes.propTypes = {
//   selectedContentType: PropTypes.string,
//   setSelectedContentType: PropTypes.func,
// };
export default ContentTypes;
