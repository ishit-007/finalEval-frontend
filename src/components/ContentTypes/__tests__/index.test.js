import ContentTypes from '..';
import renderer from 'react-test-renderer';
import React from 'react';

describe('ContentTypes', () => {
  it('should render the ContentTypes component', async () => {
    const tree = renderer.create(<ContentTypes />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});