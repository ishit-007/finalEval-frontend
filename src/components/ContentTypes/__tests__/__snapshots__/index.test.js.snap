// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentTypes should render the ContentTypes component 1`] = `
<div
  className="content-types"
>
  <div
    className="heading"
  >
    <h1
      onClick={[Function]}
    >
      Content Types
    </h1>
  </div>
  <div
    className="modal"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="modal-content"
    >
      <h3>
        Create a New Content Type
      </h3>
      <div
        className="new-content"
      >
        <p>
          Name of the content type
        </p>
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="btns"
      >
        <button
          className="cancel"
          onClick={[Function]}
        >
          Cancel
        </button>
        <button
          className="create"
          onClick={[Function]}
        >
          Create
        </button>
      </div>
    </div>
  </div>
  <div
    className="modal2"
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="modal-content2"
    >
      <h3>
        Create a New Field
      </h3>
      <div
        className="new-content"
      >
        <p>
          Name of the field
        </p>
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
      <div
        className="btns"
      >
        <button
          className="cancel"
          onClick={[Function]}
        >
          Cancel
        </button>
        <button
          className="create"
          onClick={[Function]}
        >
          Create
        </button>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="new-types-content"
    >
      <div
        className="new-types-header"
      >
        <p>
          0
           Types
        </p>
        <img
          alt=""
          src="icon-search-dark.png"
        />
      </div>
      <div
        className="new-types"
      >
        <div
          className="addNewType"
          onClick={[Function]}
        >
          + New Type
        </div>
        <div
          className="existing-types"
        />
      </div>
      <div
        className="single-type-fields"
      />
    </div>
    <div
      className="one-type-fields"
    >
      <div
        className="heading"
      >
        <h1 />
        <h3>
           Fields
        </h3>
      </div>
      <div
        className="addField"
        onClick={[Function]}
      >
        Add another field
      </div>
      <div
        className="existing-fields"
      />
    </div>
  </div>
</div>
`;
